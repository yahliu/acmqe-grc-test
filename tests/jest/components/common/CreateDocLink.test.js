/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import { createDocLink } from '../../../../src-web/components/common/CreateDocLink'

describe('createDocLink view', () => {
  const handleCreatePolicy = jest.fn()
  it('renders show create link as expected', () => {
    const component = createDocLink('en-US', handleCreatePolicy, 'show create link')
    expect(component).toMatchSnapshot()
  })

  it('renders does not show create link as expected', () => {
    const component = createDocLink('en-US', handleCreatePolicy, 'does not show create link')
    expect(component).toMatchSnapshot()
  })
})
