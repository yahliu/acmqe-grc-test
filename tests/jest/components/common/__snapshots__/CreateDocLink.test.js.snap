// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDocLink view renders does not show create link as expected 1`] = `
<div
  className="button-group"
>
  <div
    className="button-group-create-resource"
  >
    <mockTooltip
      content="You are not authorized to complete this action. See your cluster administrator for role-based access information."
      zIndex={10001}
    >
      <div
        className="disable-tooltip"
      >
        <Button
          id="create-resource"
          isDisabled={true}
          isSmall={true}
          onClick={[MockFunction]}
          variant="primary"
        >
          does not show create link
        </Button>
      </div>
    </mockTooltip>
  </div>
  <div
    className="button-group-doc-link"
  >
    <Button
      aria-describedby="launchWindow"
      component="a"
      href="https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/undefined/html/security/security#manage-security-policies"
      id="doc-link"
      isInline={true}
      target="doc"
      variant="link"
    >
      View documentation
    </Button>
  </div>
</div>
`;

exports[`createDocLink view renders show create link as expected 1`] = `
<div
  className="button-group"
>
  <div
    className="button-group-create-resource"
  >
    <mockTooltip
      content="You are not authorized to complete this action. See your cluster administrator for role-based access information."
      zIndex={10001}
    >
      <div
        className="disable-tooltip"
      >
        <Button
          id="create-resource"
          isDisabled={true}
          isSmall={true}
          onClick={[MockFunction]}
          variant="primary"
        >
          show create link
        </Button>
      </div>
    </mockTooltip>
  </div>
  <div
    className="button-group-doc-link"
  >
    <Button
      aria-describedby="launchWindow"
      component="a"
      href="https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/undefined/html/security/security#manage-security-policies"
      id="doc-link"
      isInline={true}
      target="doc"
      variant="link"
    >
      View documentation
    </Button>
  </div>
</div>
`;
